File: web/documentserver-example/ruby/app/models/file_utility.rb
# frozen_string_literal: true

#
# (c) Copyright Ascensio System SIA 2025
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

require_relative '../format/format'

# Determination file type based on extensions, utilizing `@format_manager` for format management.
class FileUtility
  @format_manager = FormatManager.new

  class << self
    attr_reader :format_manager
  end

  def self.get_file_type(file_name)
    ext = File.extname(file_name).downcase

    return 'diagram' if FileUtility.format_manager.diagram_extensions.include?(ext)
    return 'pdf' if FileUtility.format_manager.pdf_extensions.include?(ext)
    return 'word' if FileUtility.format_manager.document_extensinons.include?(ext)
    return 'cell' if FileUtility.format_manager.spreadsheet_extensinons.include?(ext)
    return 'slide' if FileUtility.format_manager.presentation_extensinons.include?(ext)

    'word'
  end
end

File: web/documentserver-example/java-spring/src/main/java/com/onlyoffice/integration/documentserver/util/file/FileUtility.java
/**
 *
 * (c) Copyright Ascensio System SIA 2021
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.onlyoffice.integration.documentserver.util.file;

import com.onlyoffice.integration.documentserver.models.enums.DocumentType;

import java.nio.file.Path;
import java.util.List;

public interface FileUtility {
    DocumentType getDocumentType(String fileName);
    String getFileName(String url);
    String getFileNameWithoutExtension(String url);
    String getFileExtension(String url);
    String getInternalExtension(DocumentType type);
    List<String> getFileExts();
    List<String> getViewedExts();
    List<String> getEditedExts();
    List<String> getConvertExts();
    Path generateFilepath(String directory, String fullFileName);
    long getMaxFileSize();
}
